packages/compass-indexes/src/components/search-indexes-table/search-indexes-table.tsx

```diff
@@ -118,14 +125,33 @@ const searchIndexFieldStyles = css({
   gap: spacing[1],
 });
 
-function SearchIndexDetails({
-  indexName,
-  definition,
-}: {
-  indexName: string;
-  definition: Document;
-}) {
+function VectorSearchIndexDetails({ definition }: { definition: Document }) {
+  return (
+    <>
+      {!definition.fields || definition.fields.length === 0
+        ? '[empty]'
```

Comment:
Is this in the designs? Should we just not allow expanding when there is nothing to show?

![image](https://github.com/mongodb-js/compass/assets/5036933/c31e14b3-b5ac-46ee-b4d9-2c4595d48cf2)

Reply:
No, it's not in the designs, however it shouldn't be something that's ever encountered. Trying to create a vector search index without any fields results in an error:
<img width="563" alt="Screenshot 2024-02-05 at 1 07 08 PM" src="https://github.com/mongodb-js/compass/assets/1791149/4545ff5e-bd4d-4380-9f95-830b13c49bed">
I don't think it's necessarily worth removing the empty case although we could remove the expanding altogether like you suggested. Since it shouldn't ever happen I think I'd prefer to just leave it as is.

Reply:
One of the indexes that you created has this state, how did we manage to create it if server should throw an error? Why are we not showing an error in this case? My concern is [empty] looks like an unexplainable placeholder that will be confusing for the users and we already handle it differently for normal indexes, so why are we breaking the pattern? If this is a totally unexpected case, why are we handling this like that?

Reply:
Ah I was wrong that's an error from something else, it can happen. Also it's worth mentioning this is also what we show when there are no mappings or fields in the mappings for regular search indexes.
How does this look instead?
<img width="1165" alt="Screenshot 2024-02-06 at 12 00 20 PM" src="https://github.com/mongodb-js/compass/assets/1791149/0cbde2bc-3d59-48c4-b22c-735660cf7d20">

Also created slack thread: https://mongodb.slack.com/archives/C05C8MVJJQ7/p1707239370305669
Thanks for bringing this up!

Reply:
Nice, this covers it very well, thanks for handling this!